From bbee3a4f806323e3e314258f42ae68232f4b80e7 Mon Sep 17 00:00:00 2001
From: hololeap <hololeap@users.noreply.github.com>
Date: Thu, 22 Sep 2022 13:37:00 -0600
Subject: [PATCH] Add flag to toggle building executable

Some distributors need to use the old Cabal-v1 build systems, which
will build the 'extensions' executable even if it isn't needed (for
instance when this package is only needed as a dependency for
hls-stan-plugin).

This adds a new 'executable' flag (enabled by default) which will allow
for disabling the 'extensions' executable if it isn't needed.

Signed-off-by: hololeap <hololeap@users.noreply.github.com>
---
 stan.cabal | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/stan.cabal b/stan.cabal
index f049259..bbb0e99 100644
--- a/stan.cabal
+++ b/stan.cabal
@@ -66,6 +66,11 @@ common common-relude
                      , relude (Relude as Prelude)
                      , relude
 
+flag executable
+  description: Build the stan executable
+  default: True
+  manual: True
+
 library
   import:              common-options
                      , common-relude
@@ -142,6 +147,8 @@ library
 
 executable stan
   import:              common-options
+  if !flag(executable)
+    buildable: False
   hs-source-dirs:      app
   main-is:             Main.hs
   build-depends:       stan
-- 
2.35.1

